manager/state/raft/raft.go

```diff
@@ -1206,10 +1206,82 @@ func (n *Node) applyRemoveNode(cc raftpb.ConfChange) (err error) {
 	// If the node from where the remove is issued is
 	// a follower and the leader steps down, Campaign
 	// to be the leader.
-	if cc.NodeID == n.Leader() {
+
+	members := n.cluster.Members()
+	if cc.NodeID == n.Leader() && !n.IsLeader() {
 		if err = n.Campaign(n.Ctx); err != nil {
 			return err
 		}
+	} else if cc.NodeID == n.Config.ID && n.IsLeader() && len(members) == 2 {
+		// if there are only 2 nodes in the cluster, and leader is leaving
+		// before closing the connection, leader has to ensure that follower gets
+		// noticed about this raft conf change commit. Otherwise, follower would
+		// assume there are still 2 nodes in the cluster and won't get elected
+		// into the leader by acquiring the majority (2 nodes)
+		//
+		// for safety, it blocks until ack is sent successfully
+		// TODO(Runshen Zhu): use leadership transfer to solve this case, after vendoring raft 3.0+
+
+		status := n.Node.Status()
+		var (
+			conn   *membership.Member
+			peerID uint64
+		)
+
+		// get peer
+		for id, member := range members {
+			if id == n.Config.ID {
+				continue
+			}
+
+			conn = member
+			break
+		}
+
+		if conn == nil {
+			n.Config.Logger.Warningf("can't find follower %x in the member list", peerID)
+			return n.cluster.RemoveMember(cc.NodeID)
+		}
+
+		// construct msg to inform peer of the commit of raft conf change
+		m := raftpb.Message{Type: raftpb.MsgApp,
+			To: conn.RaftID, From: n.Config.ID, Term: status.Term, LogTerm: status.Term,
+			Index: status.Commit, Commit: status.Commit}
+
+		ctx := context.Background()
+		backoff := 0
+		sleepTime := 10 * time.Millisecond
```

Comment:
Agreed with aaron, would be best if we relied on gRPC's backoff mechanism instead :)